app/views/granules/download.sh.erb

```diff
@@ -1,20 +1,88 @@
+        exit
+    fi
+
     fetch_urls() {
-        while read -r line; do curl -Og $line; done;
+        while read -r line; do curl -b ~/.asf_cookies -Og -L $line; done;
     }
+elif command -v wget >/dev/null 2>&1
+then
+    if [ ! -f ~/.asf_cookies ]; then
+        echo "\n.asf_cookies file not found. Please configure your username and password for authentication using a .asf_cookies file:\n    > cd ~\n    > wget --keep-session-cookies --save-cookies ~/.asf_cookies --post-data='userid=<uid>&password=<password>' --no-check-certificate https://vertex.daac.asf.alaska.edu/services/authentication\n    > chmod 0600 .asf_cookies\nwhere <uid> is your Earthdata Login username and <password> is your Earthdata Login password."
+        exit
+    fi
 
+    fetch_urls() {
+        while read -r line; do wget --load-cookies ~/.asf_cookies --save-cookies ~/.asf_cookies --keep-session-cookies $line; done;
+    }
+else
+    fail_with_error "Error: Could not find a command-line downloader.  Please install curl or wget"
+fi
+
+<% else %>
+
+if command -v curl >/dev/null 2>&1
+then
+    STATUS_CODE=`curl -s -z "$(date)" -w %{http_code} <%= raw @urls.first %> | tail -1`
+    if [ "$STATUS_CODE" -ne "200" || "$STATUS_CODE" -ne "304" ]; then
+        # Need URS authentication
+        if [ ! -f ~/.netrc ]; then
+            echo "\n.netrc file not found. Please configure your username and password for authentication using a .netrc file:\n    > echo 'default login <uid> password <password>' > .netrc\nwhere <uid> is your Earthdata Login username and <password> is your Earthdata Login password."
+            exit
+        fi
+
+        APPROVED_STATUS=`curl -s -b ~/.urs_cookies -c ~/.urs_cookies -L --max-redirs 2 -n <%= raw @urls.first %> -w %{http_code} | tail  -1`
+        if [ "$APPROVED_STATUS" -ne "302" ]; then
+            # User didn't approve the app. Direct users to approve the app in URS
+            echo "\nERROR: this is your first time retrieving data from this provider using Earthdata Login credentials. You will need to authorize the provider's web service (application) in Earthdata Login GUI before running this script again."
+            exit
+        else
+            fetch_urls() {
+                while read -r line; do curl -b ~/.urs_cookies -c ~/.urs_cookies -L -n -Og $line; done;
+            }
+        fi
+    else
+        fetch_urls() {
+            while read -r line; do curl -Og $line; done;
+        }
+    fi
 elif command -v wget >/dev/null 2>&1
 then
+    # We can't use wget to poke provider server to get info whether or not URS was integrated without download at least one of the files.
+    echo "\nWARNING: Can't find curl, use wget instead, which may affect data retrieval that authenticates through Earthdata Login."
```

Comment:
See comment in the script. ```wget``` has to download the file once to compare the size and timestamp of the local file against the remote file.